integration_test.go

```diff
@@ -60,6 +61,20 @@ func (srv *serverlessAgent) HandleBundle(w http.ResponseWriter, req *http.Reques
 		body = nil
 	}
 
+	var root Root
+	err = json.Unmarshal(body, &root)
+	if err != nil {
+		log.Printf("ERROR: failed to unmarshal serverless agent spans request body: %s", err.Error())
+	} else {
+		if len(root.Spans) > 0 {
+			if root.Spans[0].Data.SDKCustom.Tags.ReturnError == "true" ||
+				root.Spans[0].Data.Lambda.ReturnError == "true" {
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+		}
+	}
```

Comment:
Suggestion: Just combined the if checks.
```suggestion
if err != nil {
		log.Printf("ERROR: failed to unmarshal serverless agent spans request body: %s", err.Error())
	} else {
		if len(root.Spans) > 0 && (root.Spans[0].Data.SDKCustom.Tags.ReturnError == "true" ||
			root.Spans[0].Data.Lambda.ReturnError == "true") {

			w.WriteHeader(http.StatusInternalServerError)
			return
		}
	}
```